notebook/nbextensions.py

```diff
@@ -299,118 +412,356 @@ def _config_file_name_default(self):
-                if os.path.exists(nbext):
-                    print("Notebook extensions in %s:" % nbext)
-                    for ext in os.listdir(nbext):
-                        print(u"    %s" % ext)
+            sys.exit('Please specify an nbextension to install')
         else:
             try:
                 self.install_extensions()
             except ArgumentConflict as e:
-                print(str(e), file=sys.stderr)
-                self.exit(1)
+                self.log.info(str(e), file=sys.stderr)
+                sys.exit(1)
 
+_uninstall_flags = {
+    "debug" : ({
+        "UninstallNBExtensionApp" : {
+            "verbose" : 2,
+        }}, "Extra output"
+    ),
+    "quiet" : ({
+        "UninstallNBExtensionApp" : {
+            "verbose" : 0,
+        }}, "Minimal output"
+    ),
+    "user" : ({
+        "UninstallNBExtensionApp" : {
+            "user" : True,
+        }}, "Install to the user's Jupyter directory"
+    ),
+    "sys-prefix" : ({
+        "UninstallNBExtensionApp" : {
+            "sys_prefix" : True,
+        }}, "Use sys.prefix as the prefix for installing nbextensions"
+    ),
+    "py" : ({
+        "UninstallNBExtensionApp" : {
+            "python" : True,
+        }}, "Install from a Python package (alias for --python)"
+    ),
+}
+_uninstall_flags['python'] = _uninstall_flags['py']
 
-class EnableNBExtensionApp(JupyterApp):
-    name = "jupyter nbextension enable"
+class UninstallNBExtensionApp(BaseNBExtensionApp):
+    """Entry point for uninstalling notebook extensions"""
     version = __version__
-    description = "Configure an nbextension to be automatically loaded"
+    description = """Uninstall Jupyter notebook extensions
+    
+    Usage
+    
+        jupyter nbextension uninstall path/url path/url/entrypoint
+        jupyter nbextension uninstall --py pythonPackageName
+    
+    This uninstalls an nbextension.
+    """
+    
+    examples = """
+    jupyter nbextension uninstall dest/dir dest/dir/extensionjs
+    jupyter nbextension uninstall --py extensionPyPackage
+    """
+    aliases = {'section': 'ToggleNBExtensionApp.section'}
+    flags = _uninstall_flags
+    
+    user = Bool(False, config=True, help="Whether to do a user install")
+    sys_prefix = Bool(False, config=True, help="Use the sys.prefix as the prefix")
 
-    section = Unicode('notebook', config=True,
-          help=("Which config section to add the extension to. "
-                "'common' will affect all pages.")
+    prefix = Unicode('', config=True, help="Installation prefix")
+    nbextensions_dir = Unicode('', config=True, help="Full path to nbextensions dir (probably use prefix or user)")
+    destination = Unicode('', config=True, help="Destination for the copy or symlink")
+    python = Bool(False, config=True, help="Uninstall nbextension in a Python package")
+    verbose = Enum((0,1,2), default_value=1, config=True,
+        help="Verbosity level"
```

Comment:
I agree, it is weird.  I have no idea where @ellisonbg found this nor if it's actually a convention.  Honestly it looks like it really is binary, maybe log should just be a bool?